### PR TITLE
op-e2e: Add FundedAccount so parallel tests can share a single system

### PR DESCRIPTION
**Pull request by ajsutton**
**Description**

Add a `System.FundedAccount` method that creates a new, unique private key and transfers funds to it.  This allows multiple tests to run in parallel against the same System without getting nonce conflicts.

### CODE REVIEW

File: op-e2e/faultproofs/output_alphabet_test.go [hunk 1]
```diff
@@ -104,11 +104,11 @@ func TestOutputAlphabetGame_ValidOutputRoot(t *testing.T) {
 func TestChallengerCompleteExhaustiveDisputeGame(t *testing.T) {
 	op_e2e.InitParallel(t, op_e2e.UseExecutor(1))
 
-	testCase := func(t *testing.T, isRootCorrect bool) {
-		ctx := context.Background()
-		sys, l1Client := startFaultDisputeSystem(t)
-		t.Cleanup(sys.Close)
+	ctx := context.Background()
+	sys, l1Client := startFaultDisputeSystem(t)
+	t.Cleanup(sys.Close)
```
Comment (ajsutton) [inline op-e2e/faultproofs/output_alphabet_test.go:109]: I'm not sure of the exact go test life cycle but I worry a little bit that we may wind up spinning up this system, then the test pauses when we call `t.Run` and the system is left running while we're queued waiting on other tests.

Comment (trianglesphere) [reply op-e2e/faultproofs/output_alphabet_test.go:109]: Cleanup operates on the test or subtest level: https://pkg.go.dev/testing#T.Cleanup

Comment (ajsutton) [reply op-e2e/faultproofs/output_alphabet_test.go:109]: I'm more worried about the timing.  ie it runs this and starts the system, then the `t.Run` calls queue up the subtests, but potentially those subtests don't run immediately and are queued. Meanwhile the system is still running in the background waiting for the subtests to complete.

**Comment by ajsutton** [comment]
I'm not going to get back to this anytime soon - closing for now.  It will live on in history if anyone else wants to pick it up.

**Pull request closed without merging**